module/framework.js

```diff
@@ -576,44 +578,46 @@ async function _getCategories({cache=true}={}) {
     id: CATEGORY_ID.uncategorized,
     title: game.i18n.localize('WA.CategoryUncategorized'),
     position: 9e9,
-    copyForSort: [],
-    children : [],
-    parent: root,
+    articles: [],
+    articleIds: [],
+    children: [],
+    childrenIds: [],
     isUncategorized: true
   };
   categories.set(uncategorized.id, uncategorized);
 
   // Retrieve categories from the World Anvil API (build map)
   const request = await anvil.getCategories();
-  for ( let c of (request?.categories || []) ) {
+  
+  // First loop : add id listings and store in map
+  const reqCategories = (request?.categories || []);
+  for ( let c of reqCategories ) {
     categories.set(c.id, c);
-    c.copyForSort = c.children?.categories ?? [];
+    c.articleIds = (c.articles ?? []).map( a => a.id );
+    c.childrenIds = (c.children ?? []).map( ch => ch.id );
+    c.articles = [];
     c.children = [];
-    c.parentCategoryId = c.parent?.id;
-    c.parent = undefined;
-    c.folder = undefined;
   }
-  // Append children
-  for( let c of (request?.categories || []) ) {
-    const parentId = c.parentCategoryId ?? CATEGORY_ID.root;
-    const parent = categories.get(parentId) ?? root;
-    c.parent = parent;
-    parent.children.push(c);
-  }
-  // Sort children
-  for( let c of categories.values() ) {
-
-    c.children.sort( (a,b) => {
-      const indexA = c.copyForSort.findIndex( cc => cc.id === a.id );
-      const indexB = c.copyForSort.findIndex( cc => cc.id === b.id );
-      const substr = indexA - indexB;
-      if( substr != 0 ) { return substr; }
-      return a.title.localeCompare(b.title);
+
+  // Second loop : Set category.children
+  for ( let c of reqCategories ) {
+    c.children = c.childrenIds.map( id => categories.get(id) );
+    c.children.forEach( child => {
+      child.parent = c;
     });
```

Comment:
Instead of iterating twice here (`.map` and then `.forEach`) this can be handled as a side-effect inside the map:
```suggestion
    c.children = c.childrenIds.map(id => {
      const child = categories.get(id);
      child.parent = c;
      return child;
    );
```